### ISSUE TITLE
Add support for Auth (authentication) emulator

### ISSUE THREAD

**Issue by penguisnt** [cachapa/firedart#123]
Currently there is not support for the Auth emulator, although there is support for the Firestore emulator. 

This means that testing locally when you have storage security rules won't work, or you'll need to disable the rules.

It also means that login/logout always uses the production endpoints and the user IDs wont match any local Firestore data.

**Comment by penguisnt** [comment]
Here is the changes I made locally to add support for auth (hardcoded localhost and port 9099), feel free to take as much of this as you want for yourself and merge it in, I unfortunately do not have the bandwidth for tests+pull request.

Note this also includes fixes for Firestore .stream issues

https://github.com/penguisnt/firedart

**Comment by cachapa** [comment]
This is great, thanks!

Looking over your changes, they seem to make sense but they're also a single commit and a bit hard to follow the context for the changes.

I understand the lack of bandwidth but this repository depends on community contributions so I would ask if you could at least start a couple of organized PRs that make it easier to take over?
I think it would also make sense to attribute those changes to you.

I personally no longer use Firebase myself, but I keep maintaining this project since it still fills a niche in the Dart ecosystem.
For that, I hope that users of the package help fill the missing gaps so that every one benefits from it.
